build-push/bin/main.sh

```diff
@@ -0,0 +1,133 @@
+ARCHES="${ARCHES:-amd64,ppc64le,s390x,arm64}"
+
+# First arg (REPO_URL) is the clone URL for repository ('main' branch only).
+REPO_URL="$1"
+REPO_NAME=$(basename "${REPO_URL%.git}")
+# Second arg (CTX_SUB) is the context subdirectory relative to the clone path
+CTX_SUB="$2"
+# Basename of second arg names the image contents
+CTX_NAME=$(basename "$CTX_SUB")
+_REG="quay.io"
+if [[ "$REPO_NAME" =~ testing ]]; then
+    _REG="example.com"
+fi
+REPO_FQIN="$_REG/$REPO_NAME/$CTX_NAME"
+req_env_vars REPO_URL REPO_NAME CTX_SUB CTX_NAME
+
+# shellcheck disable=SC2154
+dbg "$SCRIPT_FILENAME operating constants:
+    REPO_URL=$REPO_URL
+    REPO_NAME=$REPO_NAME
+    CTX_SUB=$CTX_SUB
+    CTX_NAME=$CTX_NAME
+    REPO_FQIN=$REPO_FQIN
+"
+
+# Set non-zero to avoid actually executing build-push, simply print
+# the command-line that would have been executed
+DRYRUN=${DRYRUN:0}
+_DRNOPUSH=""
+if ((DRYRUN)); then
+    _DRNOPUSH="--nopush"
+    warn "Operating in dry-run mode with $_DRNOPUSH"
+fi
+
+### MAIN
+
+cd "$BPTMP"
+dbg "Cloning $REPO_URL"
+git clone --no-local --no-hardlinks --depth 1 --single-branch --no-tags "$REPO_URL"
+cd $REPO_NAME
+head_sha=$(git rev-parse HEAD)
+dbg "HEAD is $head_sha"
+# Labels to add to all images
+# N/B: These won't show up in the manifest-list itself, only it's constituents.
+lblargs="\
+    --label=org.opencontainers.image.source=$REPO_URL \
+    --label=org.opencontainers.image.revision=$head_sha \
+    --label=org.opencontainers.image.created=$(date -u --iso-8601=seconds)"
+dbg "lblargs=$lblargs"
+
+# tag_version.sh is sensitive to this value if set
+export img_cmd_version=""
+
+# For stable images, the version number of the command is needed for tagging.
+if [[ "$CTX_NAME" == "stable" ]]; then
+    # only native arch is needed to extract the version
+    dbg "Building local-arch image to extract stable version number"
+    podman build -t $REPO_FQIN ./$CTX_SUB
+
+    case "$REPO_NAME" in
+        skopeo) version_cmd="--version" ;;
+        buildah) version_cmd="buildah --version" ;;
+        podman) version_cmd="podman --version" ;;
+        testing) version_cmd="cat FAKE_VERSION" ;;
+        *) die "Unknown/unsupported repo '$REPO_NAME'" ;;
+    esac
+
+    pvcmd="podman run -i --rm $REPO_FQIN $version_cmd"
+    dbg "Extracting version with command: $pvcmd"
+    version_output=$($pvcmd)
+    dbg "version output:
+    $version_output
+    "
+    img_cmd_version=$(awk -r -e '/^.+ version /{print $3}' <<<"$version_output")
+    dbg "parsed version: $img_cmd_version"
+    test -n "$img_cmd_version"
+    lblargs="$lblargs --label=org.opencontainers.image.version=$img_cmd_version"
+    # Prevent temporary build colliding with multi-arch manifest list (built next)
+    # but preserve image (by ID) for use as cache.
+    dbg "Un-tagging $REPO_FQIN"
+    podman untag $REPO_FQIN
+
+    # Stable images get pushed to 'containers' namespace as latest & version-tagged
+    # Common library defines SCRIPT_PATH
```

Comment:
I don't understand this comment. Did you mean to apply this to line 47 instead of this one?

Reply:
Maybe, I can move it.  And it seems disabling SC2154 here isn't needed anyway, so I'll nuke it.